core/src/main/java/org/hibernate/ogm/loader/impl/OgmLoader.java

```diff
@@ -488,10 +551,45 @@ private TupleAsMapResultSet getResultSet(Serializable id, OgmLoadingContext ogmL
 		final TupleAsMapResultSet resultset = new TupleAsMapResultSet();
 		if ( getEntityPersisters().length > 0 ) {
 			OgmEntityPersister persister = getEntityPersisters()[0];
-			final EntityKey key = EntityKeyBuilder.fromPersister( persister, id, session );
-			Tuple entry = gridDialect.getTuple( key, persister.getTupleContext() );
-			if ( entry != null ) {
-				resultset.addTuple( entry );
+			if ( useBatching( qp ) ) {
+				// here we expect to receive QueryParameters.positionalParameters full of ids and thus of the same type.
+				// if that's not the case, we are in a bit of a trouble :)
+				// TODO Adding a safeguard but we pay the price for each call, should we remove it?
```

Comment:
In the ORM case, the query loading the batch of entities might have additional parameters like a discriminator or a @Where clause restriction I think. That's why the `QueryParameters` allows for heterogenous types. In OGM cases at least for now we don't make use of the heterogeneity.

I don't mind removing the check.